auth/src/main/java/org/aerogear/mobile/auth/credentials/JwksManager.java

```diff
@@ -108,30 +108,35 @@ public void fetchJwks(@NonNull final KeycloakConfiguration keycloakConfiguration
         HttpResponse response = getRequest.execute();
         response.onComplete(() -> {
             JsonWebKeySet jwks = null;
-            JwksException error = null;
             // this is invoked on a background thread.
-            if (response.getStatus() == 200) {
-                String jwksContent = response.stringBody();
-                try {
-                    jwks = new JsonWebKeySet(jwksContent);
-                } catch (JoseException e) {
-                    jwks = null;
-                    error = new JwksException(e);
-                    logger.warning("failed to parse JWKS key content. content = " + jwksContent);
+            try {
+                if (response.getStatus() == 200) {
+                    String jwksContent = response.stringBody();
+                    try {
+                        jwks = new JsonWebKeySet(jwksContent);
+                    } catch (JoseException e) {
+                        logger.warning("failed to parse JWKS key content. content = "
+                            + jwksContent);
+                        if(callback != null) {
+                            callback.onError(new JwksException(e));
+                        }
+                    }
+                    if (jwks != null) {
+                        persistJwksContent(keycloakConfiguration.getRealmName(), jwksContent);
```

Comment:
Put this inside the try/catch block, and you won't need to check if `jwks` is null